alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/MessageResult.java

```diff
@@ -22,21 +22,63 @@
  */
 package com.synopsys.integration.alert.common.message.model;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.AlertFieldStatus;
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.FieldStatusSeverity;
 import com.synopsys.integration.alert.common.rest.model.AlertSerializableModel;
 
 public class MessageResult extends AlertSerializableModel {
-    private String statusMessage;
+    private final String statusMessage;
+    private final List<AlertFieldStatus> fieldStatuses;
```

Comment:
Should this field be an instance of JobFieldStatuses? and the helper methods moved into the JobFieldStatuses class?

Reply:
JobFieldStatues would be a sub-set of the use case intended for this class.